index.js

```diff
@@ -724,237 +495,227 @@ class XiaomiRoborockVacuum {
   }
 
   async getSerialNumber() {
-    await this.ensureDevice("getSerialNumber");
+    await this.ensureDevice('getSerialNumber');
 
     try {
-      const serial = await this.device.call("get_serial_number");
-      this.log.info(
-        `INF getSerialNumber | ${this.model} | Serial Number is ${serial[0].serial_number}`
-      );
+      const serial = await this.device.call('get_serial_number');
+      this.log.info(`INF getSerialNumber | ${this.model} | Serial Number is ${serial[0].serial_number}`);
 
       return serial[0].serial_number;
     } catch (err) {
-      this.log.error(
-        `ERR getSerialNumber | Failed getting the firmware version.`,
-        err
-      );
+      this.log.error(`ERR getSerialNumber | Failed getting the firmware version.`, err);
       throw err;
     }
   }
 
   async getFirmware() {
-    await this.ensureDevice("getFirmware");
+    await this.ensureDevice('getFirmware');
 
     try {
-      const firmware = await this.device.call("miIO.info");
-      this.log.info(
-        `INF getFirmware | ${this.model} | Firmwareversion is ${firmware.fw_ver}`
-      );
+      const firmware = await this.device.call('miIO.info');
+      this.log.info(`INF getFirmware | ${this.model} | Firmwareversion is ${firmware.fw_ver}`);
 
       return firmware.fw_ver;
     } catch (err) {
-      this.log.error(
-        `ERR getFirmware | Failed getting the firmware version.`,
-        err
-      );
+      this.log.error(`ERR getFirmware | Failed getting the firmware version.`, err);
       throw err;
     }
   }
 
   get isCleaning() {
-    const status = this.device.property("state");
+    const status = this.device.property('state');
     return XiaomiRoborockVacuum.cleaningStatuses.includes(status);
   }
 
+
+  get isGoingToTarget() {
+    const status = this.device.property('state');
+    return 'unknown-16' === status;
```

Comment:
We can add this status in https://github.com/homebridge-xiaomi-roborock-vacuum/miio/blob/master/lib/devices/vacuum.js#L52-L88 :)

Reply:
It's probably better to ask the original MIIO author to integrate the changes (looks like the lib was not updated for quite some time). Anyway my trick seems to work. I believe state 16 is "going to target". More details: https://github.com/marcelrv/XiaomiRobotVacuumProtocol